src/nanotron/models/llama.py

```diff
@@ -45,6 +45,28 @@
 
 logger = logging.get_logger(__name__)
 
+USE_DETERMINISTIC_OPS = os.getenv("USE_DETERMINISTIC_OPS", False)
+
+# TritonRMSNorm is faster but generate randomized results.
+if USE_DETERMINISTIC_OPS is True:
+    from nanotron.nn.layer_norm import TritonRMSNorm as RMSNorm
+# Replace TritonRMSNorm with RMSNorm for a deterministic output.
+else:
+
+    class RMSNorm(nn.Module):
+        def __init__(self, hidden_size: int, eps: float = 1e-6):
+            super().__init__()
+            self.eps = eps
+            self.weight = nn.Parameter(torch.zeros(hidden_size))
+
+        def _norm(self, input: torch.Tensor) -> torch.Tensor:
+            return input * torch.rsqrt(input.pow(2).mean(-1, keepdim=True) + self.eps)
+
+        def forward(self, input: torch.Tensor) -> torch.Tensor:
+            output = self._norm(input.float())
+            output = output * (1.0 + self.weight.float())
+            return output.type_as(input)
```

Comment:
nice! did you check that this implementation of RMSNorm gives correct generations btw?
You can try to run lighteval evals with a llama2 model and compare to leaderboard